tests/rkt_auth_test.go

```diff
@@ -0,0 +1,232 @@
+	tests := []genericAuthTest{
+		{"oauth-no-config", false, "", authFailedDownload},
+		{"oauth-custom-config", true, config.DefaultCustomPath, authSuccessfulDownload},
+		{"oauth-vendor-config", true, config.DefaultVendorPath, authSuccessfulDownload},
+	}
+	testAuthGeneric(t, taas.Oauth, tests)
+}
+
+func testAuthGeneric(t *testing.T, auth taas.Type, tests []genericAuthTest) {
+	skipDestructive(t)
+	removeDataDir(t)
+	defer removeAllConfig(t)
+	server := runServer(t, auth)
+	defer server.Close()
+	for _, tt := range tests {
+		removeAllConfig(t)
+		if tt.useServerConf {
+			writeConfig(t, tt.confDir, "test.json", server.Conf)
+		}
+		expectedRunRkt(t, server.URL, tt.name, tt.line)
+	}
+}
+
+func TestAuthOverride(t *testing.T) {
+	skipDestructive(t)
+	removeDataDir(t)
+	defer removeAllConfig(t)
+	server := runServer(t, taas.Oauth)
+	defer server.Close()
+	tests := []struct {
+		vendorConfig         string
+		customConfig         string
+		name                 string
+		resultBeforeOverride string
+		resultAfterOverride  string
+	}{
+		{server.Conf, getInvalidOAuthConfig(server.Conf), "oauth-vvic", authSuccessfulDownload, authFailedDownload},
+		{getInvalidOAuthConfig(server.Conf), server.Conf, "oauth-ivvc", authFailedDownload, authSuccessfulDownload},
+	}
+	for _, tt := range tests {
+		removeAllConfig(t)
+		writeVendorConfig(t, "test.json", tt.vendorConfig)
+		expectedRunRkt(t, server.URL, tt.name+"-1", tt.resultBeforeOverride)
+		writeCustomConfig(t, "test.json", tt.customConfig)
+		expectedRunRkt(t, server.URL, tt.name+"-2", tt.resultAfterOverride)
+	}
+}
+
+func TestAuthIgnore(t *testing.T) {
+	skipDestructive(t)
+	removeDataDir(t)
+	defer removeAllConfig(t)
+	server := runServer(t, taas.Oauth)
+	defer server.Close()
+	testAuthIgnoreBogusFiles(t, server)
+	testAuthIgnoreSubdirectories(t, server)
+}
+
+func testAuthIgnoreBogusFiles(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	writeVendorConfig(t, "README", "This is vendor config")
+	writeCustomConfig(t, "README", "This is custom config")
+	writeVendorConfig(t, "test.notjson", server.Conf)
+	writeCustomConfig(t, "test.notjson", server.Conf)
+	failedRunRkt(t, server.URL, "oauth-bogus-files")
+}
+
+func testAuthIgnoreSubdirectories(t *testing.T, server *taas.Server) {
+	removeAllConfig(t)
+	customSubdir := filepath.Join(config.DefaultCustomPath, "subdir")
+	vendorSubdir := filepath.Join(config.DefaultVendorPath, "subdir")
+	writeConfig(t, customSubdir, "test.json", server.Conf)
+	writeConfig(t, vendorSubdir, "test.json", server.Conf)
+	failedRunRkt(t, server.URL, "oauth-subdirectories")
+}
+
+func runServer(t *testing.T, auth taas.Type) *taas.Server {
+	server, err := taas.NewServerWithPaths(auth, 20, "../bin/actool", "go")
+	if err != nil {
+		t.Fatalf("Could not start server: %v", err)
+	}
+	go serverHandler(t, server)
+	return server
+}
+
+func serverHandler(t *testing.T, server *taas.Server) {
+	for {
+		select {
+		case _, ok := <-server.Stop:
```

Comment:
Both are provided by taas. Listening to `Stop` channel is not strictly required as the test is fully in control of the server. Will remove it.